nullable.go

```diff
@@ -0,0 +1,39 @@
+package tfe
+
+import "encoding/json"
+
+// Nullable acts as a wrapper to enable using "omitempty" on fields that should retain the ability to explicitly be set
+// to null.
+//
+// This is particularly useful for attributes that are set to null if provided as null in a json request body.
+// Conversely, if they are omitted from the request body, they will retain their existing value.
+//
+// This struct should use generics, but this project doesn't support them yet :(
+type Nullable struct {
+	value interface{}
+}
+
+func (i Nullable) MarshalJSON() ([]byte, error) {
+	if i.value != nil {
+		return json.Marshal(i.value)
+	}
+	return json.Marshal(nil)
+}
+
+// int version of Nullable
+type NullableInt struct {
+	value *int
+}
+
+func (i NullableInt) MarshalJSON() ([]byte, error) {
+	return Nullable{i.value}.MarshalJSON()
+}
+
```

Comment:
I added a string version of nullable as well (even though it isn't in use right now) so that folks would have an easier time understanding how they might extend Nullable to work with their own types.